File: enode/src/main/java/org/enodeframework/commanding/ICommandHandlerProvider.java
package org.enodeframework.commanding;

import org.enodeframework.messaging.MessageHandlerData;

import java.util.List;

/**
 * Represents a provider to provide the command handlers.
 */
public interface ICommandHandlerProvider {
    /**
     * Get all the handlers for the given command type.
     */
    List<MessageHandlerData<ICommandHandlerProxy>> getHandlers(Class commandType);
}

File: enode/src/main/java/org/enodeframework/commanding/ICommandHandlerProvider.kt
package org.enodeframework.commanding

import org.enodeframework.messaging.MessageHandlerData

/**
 * Represents a provider to provide the command handlers.
 */
interface ICommandHandlerProvider {
    /**
     * Get all the handlers for the given command type.
     */
    fun getHandlers(commandType: Class<*>?): List<MessageHandlerData<ICommandHandlerProxy?>?>?
}